### ISSUE TITLE
Physics movement and setSpriteFrame(for animation) method issue

### ISSUE THREAD

**Issue by ilqvya** [cocos2d/cocos2d-x#17327]
- cocos2d-x version: 3.14.1
- devices test on: HP Pavilion g6 with windows10
- developing environments
   - VS version: 2015

Steps to Reproduce:
1. Start to move sprite to up or down direction by setVelocity method
2. Start sprite frame animation with Animate action or directly with: RepeatForever::create Sequence::create({  CallFunc::create([this] { setSpriteFrame(nextFrame); }, DelayTime::create(0.1f) })

The same issue in both cases. 

note: When for sprite movement uses action MoveBy, then it's all OK

the issue: Sprite with animation by setSpriteFrame or Animate action, which invoke setSpriteFrame and physics movement is shifting to the left side of screen, so the up movement moves to up and left. Left movement moves faster, right -- slower. etc....

Any idea how to solve that?

**Comment by minggo** [comment]
does the sprite's position is changed both by action and physics engine?

**Comment by ilqvya** [comment]
i don't thins so, i just swap textures with setSpriteFrame.

for now, i rewrote physics movement to action movement

**Comment by Rubonnek** [comment]
@effolkronium I also found the same issue yesterday. In my case, the animation frames differed in size due to TexturePacker cropping out the transparent pixels on the sprites. Making a call to `SpriteFrame::setOriginalSize(const Size &sizeInPixels)` to set each of the frames to the same size as they were originally before getting cropped (note that they all had the same size), and another call to `SpriteFrame::setOffset(const Vec2 &offsets)` fixed the issue for me.

**Comment by minggo** [comment]
@Rubonnek thanks for the information, so it is not a bug of engine?

**Comment by Rubonnek** [comment]
@minggo I don't think it's a bug in the physics engine. The difference in sprite frame size seems to be triggering a `cocos2d::Node` repositioning somewhere that makes its way into the physics engine and screws up with the physics simulation. Whenever I triggered my "jump" event in the game, which just sets an upwards velocity on the physics object, the sprites would jump way higher than normal depending on the animation speed and will also shift to the  left. Turns out that in my case the sprite frames differed both in height and width and by fixing the sprite frame size either by code or by editing the images, this stopped happening. Afterwards the jump movement performed as expected.
